internal/controller/chiawallet/controller.go

```diff
@@ -220,6 +222,11 @@ func (r *ChiaWalletReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	deploy := r.assembleDeployment(ctx, wallet)
+
+	if err := controllerutil.SetControllerReference(&wallet, &deploy, r.Scheme); err != nil {
+		return ctrl.Result{}, err
```

Comment:
I think this should be more like this
```suggestion
		return ctrl.Result{}, errwidsaiodsioajds
```